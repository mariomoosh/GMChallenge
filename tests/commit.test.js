import React from "react";
import renderer from "react-test-renderer";
import CommitsList from "../src/components/commits-list";
import CommitDetail from "../src/components/commit-detail";

test('CommitsList renders correctly', () => {
    const tree = renderer.create(<CommitsList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('CommitsList renders correctly', () => {
    const tree = renderer.create(<CommitDetail route={{params:{id:0,title:"home",author:"",message:""}}}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });