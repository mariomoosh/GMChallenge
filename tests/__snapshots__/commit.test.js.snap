// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitsList renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "marginTop": 0,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 40,
        "fontWeight": "bold",
      }
    }
  >
    GM Coding Challenge
  </Text>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`CommitsList renders correctly 2`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "gray",
      "padding": "20",
      "width": "20%",
    }
  }
>
  <Text>
    Hash: 
    home
  </Text>
  <Text>
    Author 
    
  </Text>
  <Text>
    Message 
    
  </Text>
</View>
`;

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "marginTop": 0,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 40,
        "fontWeight": "bold",
      }
    }
  >
    GM Coding Challenge
  </Text>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
